baler/modules/data_processing.py

```diff
@@ -158,10 +160,11 @@ def renormalize_func(norm_data: ndarray, min_list: List, range_list: List) -> nd
         ndarray: Array with the un-normalized values.
     """
     norm_data = np.array(norm_data)
-    renormalized = [
-        renormalize_std(norm_data, min_list[i], range_list[i])
```

Comment:
This code is converting the input matrix of 1000x24 size to 1000x24x1000 size, thereby creating large memory issues I believe.